anna-sala/pre-curso/airlines-pro.js

```diff
@@ -0,0 +1,258 @@
+            break;
+            }
+        }while (price !== null)
+
+    let equalPriceFlights = flights.filter(flight => flight.cost == price)
+        if (equalPriceFlights.length === 0) {
+            console.log("No hay vuelos con un precio igual a su presupuesto.");
+        } else {
+            console.log("Los vuelos con un precio igual a su presupuesto son:")
+            seeAllFlights();
+        }
+    let overPriceFlights = flights.filter(flight => flight.cost > price)
+            if (overPriceFlights.length === 0) {
+                console.log("No hay vuelos con un precio superior a su presupuesto.");
+
+            } else {
+                console.log("Los vuelos con un precio superior a su presupuesto son:")
+                seeAllFlight(overPriceFlights);
+            }
+    let underPriceFlights = flights.filter(flight => flight.cost < price)
+        if (underPriceFlights.length === 0) {
+            console.log("No hay vuelos con un precio menor a su presupuesto.");
+
+        } else {
+            console.log("Los vuelos con un precio menor a su presupuesto son:")
+            seeAllFlight(underPriceFlights);
+        }
+
+    let buyFlights = prompt("Si desea comprar alguno de los velos anteriores, escriba 'comprar'." );
+        if (buyFlights==="comprar" || buyFlights==="Comprar"){
+            buyFlightById();
+        }
+        else if (action === null) {
+            alert("Gracias por su visita, vuelva pronto!");
+            return;
+
+        } else {
+            alert ("Por favor, escriba 'comprar' para hacer una compra o presione Cancelar para salir");
+
+        } 
+}
+
+
+
+function buyFlightById(){
+    let flightId = prompt("Introduzca el ID del vuelo que desea comprar.")
+        do {
+            if (flightId === null) {
+                alert("Compra de vuelo cancelada. Gracias por usar Skylab Airlines. ");
+                return;
+        
+            } else if (flightId === "") {
+                alert("Por favor, introduzca solo el Id del vuelo");
+        
+            } else if (isNaN(flightId)) {
+                alert("Por favor, introduzca solo el Id del vuelo");
+        
+            } else {
+                flightId = Number(flightId);
+                    for (let k=0; k<flights.length; k++){
+                        if(flightId ===flights.id) {
+                            console.log("Usted ha comprado el vuelo con identificador: " + flights[i].id + " con origen " + flights[i].from + " y destino " + flights[i].to + "con un coste de " + flights[i].cost);
+                            console.log("Gracias por su compra, vuelva pronto a Skylab Airlines!")
+                        }
+                        else {
+                            alert("El ID introducido no existe. Por favor, vuelva a porbar. ")
+                            buyFlightById();
+                        }
+                    }
+        }
+        }while (flightId !== null) 
+}
```

Comment:
Tabule at line 65.

Remove 1 tabulation at line 71.

Remove break lines 27, 29, 36, 125, 145, 146, 228 and 229. Unified code styling.